source/deployment/sso-saml-adfs-msws2016.rst

```diff
@@ -0,0 +1,208 @@
+  - From the LDAP Attribute column, **select** ``E-Mail-Addresses``. From the Outgoing Claim Type, **type** ``Email``
+  - From the LDAP Attribute column, **select** ``Given-Name``. From the Outgoing Claim Type, **type** ``FirstName``
+  - From the LDAP Attribute column, **select** ``Surname``. From the Outgoing Claim Type, **type** ``LastName``
+  - From the LDAP Attribute column, **select** ``SAM-Account-Name``. From the Outgoing Claim Type, **type** ``Username``
+
+For Mattermost 3.4 and later, the *FirstName* and *LastName* attributes are optional.
+
+Then, click **Finish** to add the rule.
+
+Note that the entries in the **Outgoing Claim Type** column can be chosen to be something else. They can contain dashes but no spaces. Note that they will be used to map the corresponding fields in Mattermost later.
+
+	.. image:: ../../source/images/SSO-SAML-ADFS_create-claim-rules_003.png
+
+4. Create another new rule by clicking the **Add Rule** button.
+
+5. In the **Choose Rule Type** screen, select **Transform an Incoming Claim** from the drop-down menu, then click **Next**.
+
+	.. image:: ../../source/images/SSO-SAML-ADFS_create-claim-rules_004.png
+
+6. In the **Configure Claim Rule** screen, enter a **Claim Rule Name** of your choice, then
+  - Select *Name ID* for the **Incoming claim type**
+  - Select *Unspecified* for the **Incoming name ID format**
+  - Select *E-Mail Address* for the **Outgoing claim type**
+
+Moreover, select the **Pass through all claim values** option. Then click **Finish**.
+
+	.. image:: ../../source/images/SSO-SAML-ADFS_create-claim-rules_005.png
+
+7. Click **Finish** to create the claim rule, then **OK** to finish creating rules.
+
+8. Open Windows PowerShell as an administrator and run the following command:
+
+  ``Set-ADFSRelyingPartyTrust -TargetName <display-name> -SamlResponseSignature "MessageAndAssertion"``
+
+where <display-name> is the name you specified in step 6. In our example it would be ``mattermost``.
+
+This action will add the signature to SAML messages, making verification successful.
+
+Export Identity Provider Certificate
+-------------------------------------
+
+Next, we export the identity provider certificate, which will be later uploaded to Mattermost to finish SAML configuration.
+
+1. In ADFS management snap-in, go to **AD FS > Service > Certificates** and double click on the certificate under **Token-signing**. You may alternatively right-click the field, then click **View Certificate** in the context menu.
+
+	.. image:: ../../source/images/SSO-SAML-ADFS_export-id-provider-cert_001.png
+
+2. In the **Certificate** screen, go to the **Details** tab and click **Copy to File**, then **OK**.
+
+	.. image:: ../../source/images/SSO-SAML-ADFS_export-id-provider-cert_003.png
+
+	This opens the **Certificate Export Wizard**.
```

Comment:
I think this line can be removed as the screenshot is self-explanatory.

Reply:
agreed and removed